app/models/sales_log.rb

```diff
@@ -0,0 +1,72 @@
+class SalesLogValidator < ActiveModel::Validator
+  def validate(record); end
+end
+
+class SalesLog < Log
+  has_paper_trail
+
+  validates_with SalesLogValidator
+  before_save :update_status!
+
+  scope :filter_by_organisation, ->(org, _user = nil) { where(owning_organisation: org).or(where(managing_organisation: org)) }
+
+  STATUS = { "not_started" => 0, "in_progress" => 1, "completed" => 2 }.freeze
+  enum status: STATUS
+
+  def self.editable_fields
+    attribute_names
+  end
+
+  def form_name
+    return unless saledate
+
+    "#{collection_start_year}_#{collection_start_year + 1}_sales"
+  end
+
+  def collection_start_year
```

Comment:
This should probably also be generisized since it's essentially the same logic for both (and you'd expect to want to change them together I think)

Reply:
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/858/commits/221e4593bc39f96c7660d83050d9a4baf72dd2dc